libcron/include/libcron/Cron.h

```diff
@@ -38,11 +36,31 @@ namespace libcron
     class Cron
     {
         public:
+            template<typename WorkType>
+            bool add_schedule(std::string name, const std::string& schedule, WorkType work)
+            {
+                auto cron = CronData::create(schedule);
+                bool res = cron.is_valid();
+                if (res)
+                {
+                    tasks.lock_queue();
+                    Task t{std::move(name), CronSchedule{cron}, Task::TaskProxy(work) };
+                    if (t.calculate_next(clock.now()))
+                    {
+                        tasks.push(t);
+                    }
+                    tasks.release_queue();
+                }
+
+                return res;
+            }
 
-            bool add_schedule(std::string name, const std::string& schedule, std::function<void()> work);
             void clear_schedules();
             void remove_schedule(const std::string& name);
 
+            // Returns the delay between planned and actual execution-timepoint. 
+            std::chrono::system_clock::duration get_delay(const std::string& name);
```

Comment:
The delay is not correct until the task has been executed, so this function is really only useful in the unit tests. Since it is possible to place the check in the callback I vote to remove this function and update the tests accordingly.

Reply:
Agree - needs to be removed!

Reply:
Done!